File: src/sustech-lambda/src/main/java/cn/edu/sustc/cse/ooad/sustechlambda/sustechlambda/security/RoleAuthority.java
package cn.edu.sustc.cse.ooad.sustechlambda.sustechlambda.security;

import org.springframework.security.core.GrantedAuthority;

public class RoleAuthority implements GrantedAuthority {

    private final String roleName;

    public RoleAuthority(String roleName) {
        this.roleName = roleName;
    }

    @Override
    public String getAuthority() {
        return String.format("ROLE_%s", this.roleName);
    }
}

File: src/sustech-lambda/src/main/kotlin/cn/edu/sustc/cse/ooad/sustechlambda/security/RoleAuthority.kt
package cn.edu.sustc.cse.ooad.sustechlambda.security

import org.springframework.security.core.GrantedAuthority

class RoleAuthority(private val roleName: String) : GrantedAuthority {

    override fun getAuthority() = "ROLE_${this.roleName}"
}
